src/sync.cpp

```diff
@@ -661,24 +661,32 @@ LocalNode* Sync::checkpath(LocalNode* l, string* localpath, string* localname)
                                 handlelocalnode_map::iterator it;
+                                const char *colon;
+                                fsfp_t fp1, fp2;
 
                                 // was the file overwritten by moving an existing file over it?
-                                if ((it = client->fsidnode.find(fa->fsid)) != client->fsidnode.end())
+                                if ((it = client->fsidnode.find(fa->fsid)) != client->fsidnode.end()
+                                        && l->sync == it->second->sync
+                                            || ((fp1 = l->sync->dirnotify->fsfingerprint())
+                                                && (fp2 = it->second->sync->dirnotify->fsfingerprint())
+                                                && (fp1 == fp2)
+                                            #ifdef _WIN32
+                                                // only consider fsid matches betweek different syncs for local drives with the
+                                                // same drive letter, to prevent problems with cloned Volume IDs
+                                                && (colon = strstr(parent->sync->localroot.name.c_str(), ":"))
+                                                && !memcmp(parent->sync->localroot.name.c_str(),
+                                                       it->second->sync->localroot.name.c_str(),
+                                                       colon - parent->sync->localroot.name.c_str())
+                                            #endif
+                                                )
+                                    )
                                 {
                                     // catch the not so unlikely case of a false fsid match due to
                                     // e.g. a file deletion/creation cycle that reuses the same inode
                                     if (it->second->mtime != fa->mtime || it->second->size != fa->size)
                                     {
-                                        // do not delete the LocalNode if it is in a different filesystem
-                                        // because it could be a file with the same fsid in another drive
-                                        fsfp_t fp1, fp2;
-                                        if (l->sync == it->second->sync
-                                            || ((fp1 = l->sync->dirnotify->fsfingerprint())
-                                                && (fp2 = it->second->sync->dirnotify->fsfingerprint())
-                                                && (fp1 == fp2)))
-                                        {
-                                            delete it->second;
-                                        }
+                                        l->mtime = -1;  // trigger change detection
+                                        delete it->second;   // delete old LocalNode
```

Comment:
If a false `fsid` match is detected, not only delete the previous `LocalNode` as before, but also trigger a change detection for the current `LocalNode`, because the `fsid` doesn't match so it's needed to, at least, update it.